### ISSUE TITLE
Further Improve Handling of huge remote images

### ISSUE THREAD

**Issue by BentiGorlich** [MbinOrg/mbin#511]
Follow-up issue for #501 

Main problem after #501: if there is an entry without a local image (because it was too large) we do not have a thumb_image of that.

Solution: compress the image and safe it afterwards. We add a new column is_compressed, so we can display the original full res image from the source url when it is needed and we know that we don't have the original full res image at hand

**Comment by melroy89** [comment]
Main problem after #501: If we display a remote image (from another domain), because there is no local image, this can cause issues with the server CORS and CSP HTTP Headers. Thus now allowing to load content from other domains (due to security concerns).
